src/main/java/com/napier/sem/DatabaseConnectionImpl.java

```diff
@@ -0,0 +1,133 @@
+package com.napier.sem;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DatabaseConnectionImpl implements DatabaseConnection {
+
+    private static DatabaseConnectionImpl instance;
+    static final int RETRIES = 10;
+    /**
+     * Connection to MySQL database.
+     */
+    private Connection connection = null;
+
+    private DatabaseConnectionImpl(){
+    }
+
+    public static DatabaseConnection getInstance(){
+        if (instance == null){
+            instance = new DatabaseConnectionImpl();
+        }
+        return instance;
+    }
+
+    /**
+     * Accepts an SQL statement, executes it and returns the result of the query
+     * @param request a string in the form of an SQL statement
+     * @return List of results from the database. Returns null if there was an error.
+     */
+    @Override
+    public List<Map<String, Object>> executeSQLStatement(String request) {
+        ResultSet result = null;
+        List<Map<String, Object>> resultList = new ArrayList<Map<String, Object>>();
+        Map<String, Object> row = null;
+
+        this.connect();
+        try {
+            Statement stmt = connection.createStatement();
+            result = stmt.executeQuery(request);
+
+            ResultSetMetaData metaData = result.getMetaData();
+            Integer columnCount = metaData.getColumnCount();
```

Comment:
Why Integer and not int?

Reply:
Same q again

Reply:
Because that's what they did on StackOverflow when I looked it up. ;)